## 説明変数にサークルに所属しているか否かの情報を選択した場合のF_0値を計算

In [1]:
#ライブラリをインポート
import pandas as pd
import numpy as np
import math
import statsmodels.api as sm

In [2]:
#データフレームの作成
df = pd.read_csv("club_activity.csv", sep=",")

In [4]:
#説明変数を取り出す
x = df['x2'].values

#ターゲット変数を取り出す
y = df['score'].values

In [6]:
#モデルの作成、バイアス項を追加
model = sm.OLS(y, sm.add_constant(x))
results = model.fit()

#結果を表示
print(results.summary())

                            OLS Regression Results                            
Dep. Variable:                      y   R-squared:                       0.142
Model:                            OLS   Adj. R-squared:                  0.035
Method:                 Least Squares   F-statistic:                     1.322
Date:                Mon, 24 Aug 2020   Prob (F-statistic):              0.283
Time:                        21:14:51   Log-Likelihood:                -35.999
No. Observations:                  10   AIC:                             76.00
Df Residuals:                       8   BIC:                             76.60
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         75.2000      4.427     16.986      0.0

/opt/anaconda3/lib/python3.7/site-packages/scipy/stats/stats.py:1450: UserWarning: kurtosistest only valid for n>=20 ... continuing anyway, n=10
  "anyway, n=%i" % int(n))


今回F_0の値は1.32となっている。線形代数のF_0値が5.51であった。線形代数＋サークルで作った回帰式の偏回帰係数（19.0 vs 9.0）を
比較することで線形代数の方が目的変数への影響力は大きいとわかる。  
Linear Algebra Test & Club activityのスクリプトに戻る。  